modules/javafx.controls/src/main/java/javafx/scene/control/skin/MenuButtonSkinBase.java

```diff
@@ -144,24 +144,15 @@ public MenuButtonSkinBase(final C control) {
         };
         control.getItems().addListener(itemsChangedListener);
 
-        if (getSkinnable().getScene() != null) {
-            ControlAcceleratorSupport.addAcceleratorsIntoScene(getSkinnable().getItems(), getSkinnable());
-        }
+        ControlAcceleratorSupport.addAcceleratorsIntoScene(getSkinnable().getItems(), getSkinnable());
 
         List<Mnemonic> mnemonics = new ArrayList<>();
         sceneChangeListener = (scene, oldValue, newValue) -> {
             if (oldValue != null) {
-                ControlAcceleratorSupport.removeAcceleratorsFromScene(getSkinnable().getItems(), oldValue);
```

Comment:
will it handle a case where the menu button gets attached to a different scene?
could you add a second test for this scenario please?

And i wonder if the problem is in ControlAcceleratorSupport rather than here.  We do have a similar code in Control:380, do we have a problem there?

Reply:
The code in ControlAcceleratorSupport adds its own scene listeners for each node added. MenuButtonSkinBase does the same, unlike Control which only calls it once.

I think we could fix it in ControlAcceleratorSupport alternatively or in addition to this fix. The code in ControlAcceleratorSupport appears to do everything this would've done.

I've added the test for changing scenes.